### PR TITLE
aide-app 1.94.2.24317 (new cask)

### PR DESCRIPTION
**Pull request by aaditkamat**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [ ] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

File: Casks/a/aide-app.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  livecheck do
+    url :url
+    strategy :github_latest
+  end
+
+  depends_on macos: ">= :catalina"
+
+  app "Aide.app"
+
+  # Documentation: https://docs.brew.sh/Cask-Cookbook#stanza-zap
```
Comment (bevanjkay) [inline Casks/a/aide-app.rb:20]:
```suggestion
```

**Comment by daeho-ro** [comment]
@aaditkamat , Thanks for your contribution! I left some comments on the code, please check it.

File: Casks/a/aide-app.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  zap trash: [
+    "~/Users/aaditkamat/Library/Application Support/Aide",
+    "~/Users/aaditkamat/Library/Caches/ai.codestory.AideInsiders",
+    "~/Users/aaditkamat/Library/Caches/ai.codestory.AideInsiders.savedState",
+    "~/Users/aaditkamat/Library/Preferences/ai.codestory.AideInsiders.plist",
+    "~/Users/aaditkamat/Library/Saved Application State/ai.codestory.AideInsiders.savedState",
```
Comment (khipp) [inline Casks/a/aide-app.rb:26]:
```suggestion
    "~/Library/Application Support/Aide",
    "~/Library/Caches/ai.codestory.AideInsiders",
    "~/Library/Caches/ai.codestory.AideInsiders.savedState",
    "~/Library/Preferences/ai.codestory.AideInsiders.plist",
    "~/Library/Saved Application State/ai.codestory.AideInsiders.savedState",
```

File: Casks/a/aide-app.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
+cask "aide-app" do
```
Comment (khipp) [inline Casks/a/aide-app.rb:1]:
```suggestion
cask "aide" do
```

File: Casks/a/aide-app.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  version "1.94.2.24311"
+  sha256 "92daf0b4cf370e6519ff3a0c2ccc5df808ee15dd19cfa9a332216c2358e71fa8"
+
+  url "https://github.com/codestoryai/binaries/releases/download/#{version}/Aide.arm64.#{version}.dmg",
```
Comment (khipp) [inline Casks/a/aide-app.rb:5]:
```suggestion
  arch arm: "arm64", intel: "x64"

  version "1.94.2.24313"
  sha256 arm:   "a076384e68e161d202f2a52c9439b2e01f911c5dc15e0b347d71b3a59d989445",
         intel: "81ac2ba32ad1e78d42a709c90e435572fe2e409550fff371c964c7b112735e3a"

  url "https://github.com/codestoryai/binaries/releases/download/#{version}/Aide.#{arch}.#{version}.dmg",
```

File: Casks/a/aide-app.rb (hunk 2 above)
Comment (aaditkamat) [reply Casks/a/aide-app.rb:25]: Oops good catch! Not sure how I included both the tilde as well as the hardcoded HOME location

**Comment by khipp** [comment]
@aaditkamat You still need to rename the file to `aide.rb`. Please also squash your commits to keep the history cleaner. Thanks!

**Review by daeho-ro — APPROVED** [review]

**Comment by daeho-ro** [comment]
The name is duplicate with a formula but it seems that they are not related.

**Comment by aaditkamat** [comment]
> The name is duplicate with a formula but it seems that they are not related. How can we pass this? Maybe this is the reason why the name was `aide-app` 😅

Yes that's why I had named the cask that way. But I thought that you were ok with making the exception to the cask name being different from the file name

**Review by khipp — APPROVED** [review]
Thank you, @aaditkamat!

File: Casks/a/aide-app.rb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+  zap trash: [
+    "~/Library/Application Support/ai.codestory.sidecar",
+    "~/Library/Application Support/Aide",
+    "~/Library/Caches/ai.codestory.AideInsiders",
+    "~/Library/Caches/ai.codestory.AideInsiders.savedState",
+    "~/Library/Caches/ai.codestory.AideInsiders.ShipIt",
+    "~/Library/HTTPStorages/ai.codestory.AideInsiders",
+    "~/Library/Preferences/ai.codestory.AideInsiders.plist",
+    "~/Library/Preferences/ByHost/ai.codestory.AideInsiders.ShipIt.6B4DD3EE-2BFA-5A1C-A64F-50799C342D41.plist",
```
Comment (krehel) [inline Casks/a/aide-app.rb:37]:
```suggestion
    "~/Library/Preferences/ByHost/ai.codestory.AideInsiders.ShipIt.*.plist",
```

**Comment by krehel** [comment]
We are going to need to either rename the binary, forgo installing it, or implement `conflicts_with: aide`

I installed `aide` the formula, and then we would get this -

`Error: It seems there is already a Binary at '/opt/homebrew/bin/aide'

File: Casks/a/aide.rb [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+  livecheck do
+    url "https://aide-updates.codestory.ai/api/update/darwin-#{arch}/stable/0"
+    strategy :json do |json|
+      json["productVersion"]
+    end
+  end
+
+  auto_updates true
+  conflicts_with formula: aide
```
Comment (bevanjkay) [inline Casks/a/aide.rb:22]:
```suggestion
  conflicts_with formula: "aide"

**Review by bevanjkay — APPROVED** [review]

**Pull request merged**